packages/profile-sync-controller/src/sdk/authentication.ts

```diff
@@ -54,22 +49,13 @@ export class JwtBearerAuth implements SIWEInterface, SRPInterface {
   }
 
   async connectSnap(): Promise<string> {
-    const provider = await getMetaMaskProviderEIP6963();
-    if (!provider) {
-      throw new Error('failed to get MetaMaskProviderEIP6963 provider');
-    }
-    const res = await connectSnap(provider);
-    return res;
+    this.#assertSRP(this.#type, this.#sdk);
```

Comment:
Ensures that we keep some separation of concerns between SIWE Auth and SRP Auth classes.
We do not want SIWE AuthType from calling these snap methods.